Branch-SDK/src/io/branch/referral/indexing/BranchUniversalObject.java

```diff
@@ -0,0 +1,506 @@
+package io.branch.referral.indexing;
+
+import android.app.Activity;
+import android.content.Context;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import io.branch.referral.Branch;
+import io.branch.referral.BranchError;
+import io.branch.referral.BranchShortLinkBuilder;
+import io.branch.referral.Defines;
+import io.branch.referral.util.LinkProperties;
+import io.branch.referral.util.ShareSheetStyle;
+
+/**
+ * <p>Class represents a single piece of content within your app, as well as any associated metadata.
+ * It provides convenient methods for sharing, deep linking, and tracking how often that content is viewed. This information is then used to provide you with powerful content analytics
+ * and deep linking.BranchUniversalObject provides convenient methods to work on your content with Branch services
+ * </p>
+ */
+public class BranchUniversalObject {
+    /* Canonical identifier for the content referred. Normally the canonical path for your content in the app or web */
+    private String canonicalIdentifier_;
+    /* Title for the content referred by BranchUniversalObject */
+    private String title_;
+    /* Description for the content referred */
+    private String description_;
+    /* An image url associated with the content referred */
+    private String imageUrl_;
+    /* Meta data provided for the content. This meta data is used as the link parameters for links created from this object */
+    private final HashMap<String, String> metadata_;
+    /* Mime type for the content referred */
+    private String type_;
+    /* Content index mode */
+    private CONTENT_INDEX_MODE indexMode_;
```

Comment:
We internally represent this as boolean now. The reason for providing Enum instead of a boolean is that we don't  need to change the API  in case we have more options (Like `Public_IN_APP`)  in future